### ISSUE TITLE
No workie macosie 😭! ! !

### ISSUE THREAD

**Issue by BoogeyMan24** [formerlybased/feather-engine#59]
it doesn't work on macos (base project zero changes):
```
In file included from /Users/home/Desktop/Desktop/feather-engine2/feather-engine/feather/components/drawable.cpp:1:
In file included from /Users/home/Desktop/Desktop/feather-engine2/feather-engine/feather/components/drawable.h:5:
/Users/home/Desktop/Desktop/feather-engine2/feather-engine/feather/windows/window.h:5:10: fatal error: 'SDL2/SDL.h' file not found
#include <SDL2/SDL.h>
               ^~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/game.dir/feather/components/drawable.cpp.o] Error 1
make[1]: *** [CMakeFiles/game.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by formerlybased** [comment]
do you have SDL2 installed, and if so where

**Comment by formerlybased** [comment]
hey @BoogeyMan24 can you try it again? ideally I want it to work on all platforms

**Comment by JavinalCharles** [comment]
The following lines in CMakeLists.txt might be what:s causing the error.
```
find_library(LIB_IMAGE SDL2_image ./lib/Unix/lib_image)
find_library(LIB_MIXER SDL2_mixer ./lib/Unix/lib_mixer)
find_library(LIB_TTF SDL2_ttf ./lib/Unix/lib_ttf)

find_path(LIB_INCLUDES SDL_image.h ./include/Unix/SDL2)
```

cmake is being instructed to use the libraries for *nix version of SDL included when we clone the repository. It might be better to instruct cmake to include different versions of the library depending on users` systems. This stack overflow might help: [[OS specific instructions in CMAKE: How to?](https://stackoverflow.com/questions/9160335/os-specific-instructions-in-cmake-how-to)]

**Comment by formerlybased** [comment]
The issue is that, in theory, the unix version should be the one used on macos. Either way, this probably means it won't compile on windows either. That's something that needs to be fixed.

**Comment by formerlybased** [comment]
I wonder if we can use SDL as a git submodule instead of directly including the files in the project

**Comment by JavinalCharles** [comment]
It might have been better to just cite SDL2 as a dependency instead of including the library files. So cmake  can just use what is locally installed in the user's system.

**Comment by formerlybased** [comment]
Yeah, that's how it was super early on, my only issue with that is that it makes it harder for Windows users since it's difficult to install system-wide libraries on Windows and including SDL2 in the project makes it a bit easier to use. I'm going to see if I can find a good solution to this.

**Comment by kiner-shah** [comment]
Maybe you can try checking FetchContent_Declare or ExternalProject_Add of CMake. Submodules also is fine I think.
